bridges/snowbridge/pallets/ethereum-client/src/impls.rs

```diff
@@ -20,9 +22,13 @@ impl<T: Config> Verifier for Pallet<T> {
 			proof.block_hash,
```

Comment:
The `proof.tx_index` and `proof.block_hash` fields are not actually used in the verification (at least after the changes in this PR), and can not even be trusted. The relayer can pass any values for them.

I think we should remove them from the `Proof` struct, even if it makes debugging harder.

Note that the relayer does have the knowledge of the mapping of ethereum transaction id <-> substrate transaction id, and so that is one way we can troubleshoot in future.

At least `proof.block_hash` can be replaced with `execution_header.hash_tree_root()`.